Core/include/Acts/Surfaces/DiscSurface.hpp

```diff
@@ -48,36 +48,33 @@ class DiscSurface : public Surface {
  protected:
   /// Constructor for Discs from Transform3D, \f$ r_{min}, r_{max} \f$
   ///
-  /// @param htrans is transform that places the disc in the global 3D space
-  /// (can be nullptr)
+  /// @param transform is transform that places the disc in the global 3D space
   /// @param rmin The inner radius of the disc surface
   /// @param rmax The outer radius of the disc surface
   /// @param hphisec The opening angle of the disc surface and is optional
   ///        the default is a full disc
-  DiscSurface(std::shared_ptr<const Transform3D> htrans, double rmin,
-              double rmax, double hphisec = M_PI);
+  DiscSurface(const Transform3D& transform, double rmin, double rmax,
+              double hphisec = M_PI);
 
   /// Constructor for Discs from Transform3D, \f$ r_{min}, r_{max}, hx_{min},
   /// hx_{max} \f$
   /// This is n this case you have DiscTrapezoidBounds
   ///
-  /// @param htrans is transform that places the disc in the global 3D space
-  /// (can be nullptr)
+  /// @param transform is transform that places the disc in the global 3D space
   /// @param minhalfx The half length in x at minimal r
   /// @param maxhalfx The half length in x at maximal r
   /// @param maxR The inner radius of the disc surface
   /// @param minR The outer radius of the disc surface
   /// @param avephi The position in phi (default is 0.)
   /// @param stereo The optional stereo angle
-  DiscSurface(std::shared_ptr<const Transform3D> htrans, double minhalfx,
-              double maxhalfx, double maxR, double minR, double avephi = 0.,
-              double stereo = 0.);
+  DiscSurface(const Transform3D& transform, double minhalfx, double maxhalfx,
+              double maxR, double minR, double avephi = 0., double stereo = 0.);
```

Comment:
Why now, but this needs another PR, was it will touch the `TrapezoidDiscBounds` class.

Reply:
I just saw it then and if it would have been a small change then it would have been fine to add it in this PR. If it's larger then for sure, another PR is needed.

Reply:
I opened an issue.

Reply:
Thank you

Reply:
I opened #442 for this.